Medium/SuperUglyNumber.java

```diff
@@ -0,0 +1,29 @@
+public class Solution {
+    public int nthSuperUglyNumber(int n, int[] primes) {
+        //similar to ugly number 2, DP
+        //d[i] = Math.min(min, d[i-1]*primes[j])
+        //  example:   6 [2,3,7] 
+        //2*[1],3*[1],7*[1] compare 2,3,7
+        //[1,2]->2*[2],3*[1,2],7*[1,2] compare [4,3,7]
+        //....
+        //Time O(n*(primes.length))
+    
+        int[] d =new int[n];
```

Comment:
If I understand correctly, p is an array to remember how many times each prime has been used. May be better if you use understandable variable names, like primeCount instead of p.